### ISSUE TITLE
Ensure MSVC versions matching Python compiler are available

### ISSUE THREAD

**Issue by tobecodex** [Azure/Azure-Functions#772]
tl;dr; Make sure _MSVC_VER=9.0 & 10.0 are available to support Python2.7, 3.4 module installs. (3.6 requires v14 which is already there)

For Windows platform only. Runtime v1, Consumption Plan.

This is all in the context of trying to get a working install of spaCy (https://github.com/explosion/spaCy).

A number of Python modules require compiler support to install. The compiler version must match the one the python interpreter was compiled with. See here for those versions: https://wiki.python.org/moin/WindowsCompilers

Repro is simple, from a Kudo console, assuming you're in a virtualenv

```
python -m pip install spacy
```

For vanilla python27 you'll see lots of errors relating to insecure platform which can be safely ignored but the interesting one here is:

```
    building 'murmurhash.mrmr' extension
    error: Microsoft Visual C++ 9.0 is required (Unable to find vcvarsall.bat). Get it from http://aka.ms/vcpython27
```

So one of spaCy's deps requires a compiler to install. The standard response to this is to install from wheels which does work for all of spaCy's dependencies but, crucially, doesn't for spaCy itself. The system just appears to hang and eventually reset itself. Even with verbose flag for install there is no clue what's going on and eventually you just end up back at what looks like a fresh console with, kicked out of your venv without spaCy having been installed. Maybe a watchdog killed a CPU intensive process that had been running 'too long' ?

**Comment by tobecodex** [comment]
Update: Doesn't look like you have any compilers available for Windows in reality. Looks like just enough to do remote debugging.

Just tried to install the package I need after from a wheel having install all it's' dependencies. Here's what happened:

```
d:\home\site\wwwroot\NER>python -m pip install spacy-2.0.11-cp36-cp36m-win_amd64.whl
Processing d:\home\site\wwwroot\ner\spacy-2.0.11-cp36-cp36m-win_amd64.whl
Requirement already satisfied: thinc>=6.10.1 in d:\home\python364x64\lib\site-packages (from spacy==2.0.11) (6.10.2)
Requirement already satisfied: ujson>=1.35 in d:\home\python364x64\lib\site-packages (from spacy==2.0.11) (1.35)
Requirement already satisfied: preshed>=1.0.0 in d:\home\python364x64\lib\site-packages (from spacy==2.0.11) (1.0.0)
Requirement already satisfied: dill>=0.2 in d:\home\python364x64\lib\site-packages (from spacy==2.0.11) (0.2.7.1)
Requirement already satisfied: murmurhash>=0.28 in d:\home\python364x64\lib\site-packages (from spacy==2.0.11) (0.28.0)
Requirement already satisfied: plac>=0.9.6 in d:\home\python364x64\lib\site-packages (from spacy==2.0.11) (0.9.6)
Requirement already satisfied: numpy>=1.7 in d:\home\python364x64\lib\site-packages (from spacy==2.0.11) (1.14.2)
Requirement already satisfied: regex>=2017.4.5 in d:\home\python364x64\lib\site-packages (from spacy==2.0.11) (2018.2.21)
Requirement already satisfied: pathlib in d:\home\python364x64\lib\site-packages (from spacy==2.0.11) (1.0.1)
Requirement already satisfied: cymem>=1.30 in d:\home\python364x64\lib\site-packages (from spacy==2.0.11) (1.31.2)
Requirement already satisfied: termcolor in d:\home\python364x64\lib\site-packages (from thinc>=6.10.1->spacy==2.0.11) (1.1.0)
Requirement already satisfied: tqdm>=4.10.0 in d:\home\python364x64\lib\site-packages (from thinc>=6.10.1->spacy==2.0.11) (4.23.0)
Requirement already satisfied: msgpack-python in d:\home\python364x64\lib\site-packages (from thinc>=6.10.1->spacy==2.0.11) (0.5.6)
Requirement already satisfied: six>=1.10.0 in d:\home\python364x64\lib\site-packages (from thinc>=6.10.1->spacy==2.0.11) (1.11.0)
Requirement already satisfied: cytoolz>=0.8 in d:\home\python364x64\lib\site-packages (from thinc>=6.10.1->spacy==2.0.11) (0.9.0.1)
Requirement already satisfied: msgpack-numpy in d:\home\python364x64\lib\site-packages (from thinc>=6.10.1->spacy==2.0.11) (0.4.3)
Requirement already satisfied: wrapt in d:\home\python364x64\lib\site-packages (from thinc>=6.10.1->spacy==2.0.11) (1.10.11)
Requirement already satisfied: pyreadline>=1.7.1 in d:\home\python364x64\lib\site-packages (from dill>=0.2->spacy==2.0.11) (2.1)
Requirement already satisfied: toolz>=0.8.0 in d:\home\python364x64\lib\site-packages (from cytoolz>=0.8->thinc>=6.10.1->spacy==2.0.11) (0.9.0)
Requirement already satisfied: msgpack>=0.3.0 in d:\home\python364x64\lib\site-packages (from msgpack-numpy->thinc>=6.10.1->spacy==2.0.11) (0.5.6)
Installing collected packages: spacy

process [7728] terminated!  Press ENTER to start a new cmd process.

Microsoft Windows [Version 10.0.14393]
(c) 2016 Microsoft Corporation. All rights reserved.

D:\home>python --version
Python 3.6.4

D:\home>python -m spacy
```
Same happens for 27, btw. Couldn't see any processes running but looked like the process explorer was hanging also (CPU usage??).

**Comment by mhoeger** [comment]
I've marked this as "wontfix", as we aren't investing in V1 python support (it's an experimental language) and this is a corner case. We do have Python [on our radar for V2,](https://github.com/Azure/Azure-Functions/wiki/Language-support#what-about-the-experimental-languages-that-are-in-v1) and feedback from python users like yourself would be helpful to define our priorities! If you have feedback, we'd really love to hear your thoughts through this [survey](https://microsoft.qualtrics.com/jfe/form/SV_3OYZvwEglbDG925)!
